speechbrain/utils/logger.py

```diff
@@ -61,6 +62,118 @@
+        flag to decide whether to log based on the current process.
+
+        Parameters
+        ----------
+        level : int
+            Logging level (e.g., logging.INFO, logging.WARNING).
+        msg : str
+            The message to log.
+        *args : tuple
+            Additional positional arguments passed to the logger.
+        **kwargs : dict
+            Additional keyword arguments passed to the logger, including:
+            - main_process_only (bool): If True, log only from the main process (default: True).
+            - stacklevel (int): The stack level to use when logging (default: 2).
+
+        Notes
+        -----
+        If `main_process_only` is True, the log will only be written if the current process
+        is the main process, as determined by `if_main_process()`.
+        """
+        main_process_only = kwargs.pop("main_process_only", True)
+        kwargs.setdefault("stacklevel", 2)
+
+        if self.isEnabledFor(level):
+            if self._should_log(main_process_only):
+                msg, kwargs = self.process(msg, kwargs)
+                self.logger.log(level, msg, *args, **kwargs)
+
+    @functools.lru_cache(None)
+    def warning_once(self, *args: tuple, **kwargs: dict):
+        """
+        Logs a warning message only once by using caching to prevent duplicate warnings.
+
+        Parameters
+        ----------
+        *args : tuple
+            Positional arguments passed to the warning log.
+        **kwargs : dict
+            Keyword arguments passed to the warning log.
+
+        Notes
+        -----
+        This method is decorated with `functools.lru_cache(None)`, ensuring that the warning
+        message is logged only once regardless of how many times the method is called.
+        """
+        self.warning(*args, **kwargs)
+
+
+def get_logger(name: str, log_level: str = None) -> MultiProcessLoggerAdapter:
+    """
+    Retrieves a logger by name and optionally sets its log level based on an environment
+    variable or the provided log level.
+
+    Parameters
+    ----------
+    name : str
+        The name of the logger to retrieve.
+    log_level : str, optional
+        The logging level to set for the logger. If not provided, the function will look
+        for an environment variable `SB_LOG_LEVEL` to determine the log level.
+
+    Returns
+    -------
+    MultiProcessLoggerAdapter
+        An instance of `MultiProcessLoggerAdapter` wrapping the logger with the specified name.
+    
+    Notes
+    -----
+    If `log_level` is provided or found in the environment, both the logger and the root
+    logger will have their log levels set accordingly.
+    """
+    if log_level is None:
+        log_level = os.environ.get("SB_LOG_LEVEL", None)
+    logger = logging.getLogger(name)
+    if log_level is not None:
+        logger.setLevel(log_level.upper())
+        logger.root.setLevel(log_level.upper())
+    return MultiProcessLoggerAdapter(logger, {})
```

Comment:
Hey @asumagic @TParcollet @pplantinga,

what do you think of this new logger prototype ? The key idea is to allow to perform logging only on the main proc (you can turn down this option by setting `main_proc_only=False`) and also to only emit on one warning if wanted (which was something that we wanted to have in some previous prs). 

The drop-in replacement is quite easy as it follows the same API of `logging` with `get_logger`.

Reply:
The implementation looks reasonable enough to me. As for the environment variable it feels strange to only have it documented here, but I don't know where else to at the moment (maybe we should have a debugging tutorial/tips page in the doc sometime).

Also did you check how it interacts with how logging is configured by default in SB (i.e. whether this actually works at all)? Since for instance we have different log levels for the standard output and the recipe file output.

Reply:
> The implementation looks reasonable enough to me. As for the environment variable it feels strange to only have it documented here, but I don't know where else to at the moment (maybe we should have a debugging tutorial/tips page in the doc sometime).
I agree on this. We should have a documentation about all the tips such as using mixed precision etc. I feel that having an environment variable can be useful given the fact that modifying the log level through the python scripts might be inconvenient. 

> Also did you check how it interacts with how logging is configured by default in SB (i.e. whether this actually works at all)? Since for instance we have different log levels for the standard output and the recipe file output.
I am currently testing it :p

Reply:
I made some slight changes to remove how to setup logger level so that we only rely on `setup_logging`.